e2e/addons/addons_test.go

```diff
@@ -0,0 +1,215 @@
+				EnvName:     "test",
+				Profile:     "default",
+				Prod:        false,
+			})
+		})
+
+		It("env init should succeed", func() {
+			Expect(testEnvInitErr).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("when adding an app", func() {
+		var (
+			appInitErr error
+		)
+		BeforeAll(func() {
+			_, appInitErr = cli.AppInit(&client.AppInitRequest{
+				AppName:    appName,
+				AppType:    "Load Balanced Web App",
+				Dockerfile: "./hello/Dockerfile",
+				AppPort:    "80",
+			})
+		})
+
+		It("app init should succeed", func() {
+			Expect(appInitErr).NotTo(HaveOccurred())
+		})
+
+		It("app init should create app manifests", func() {
+			Expect("./ecs-project/hello/manifest.yml").Should(BeAnExistingFile())
+
+		})
+
+		It("app ls should list the apps", func() {
+			appList, appListError := cli.AppList(projectName)
+			Expect(appListError).NotTo(HaveOccurred())
+			Expect(len(appList.Apps)).To(Equal(1))
+
+			appsByName := map[string]client.AppDescription{}
+			for _, app := range appList.Apps {
+				appsByName[app.AppName] = app
+			}
+
+			for _, app := range []string{appName} {
+				Expect(appsByName[app].AppName).To(Equal(app))
+				Expect(appsByName[app].Project).To(Equal(projectName))
+			}
+		})
+	})
+
+	Context("copy addons file to ecs-project", func() {
+		It("should copy all addons/ files to the app's workspace", func() {
+			err := os.MkdirAll("./ecs-project/hello/addons", 0777)
+			Expect(err).NotTo(HaveOccurred(), "create addons dir")
+
+			fds, err := ioutil.ReadDir("./hello/addons")
+			Expect(err).NotTo(HaveOccurred(), "read addons dir")
+
+			for _, fd := range fds {
+				destFile, err := os.Create(fmt.Sprintf("./ecs-project/hello/addons/%s", fd.Name()))
+				Expect(err).NotTo(HaveOccurred(), "create destination file")
+				defer destFile.Close()
+
+				srcFile, err := os.Open(fmt.Sprintf("./hello/addons/%s", fd.Name()))
+				Expect(err).NotTo(HaveOccurred(), "open source file")
+				defer srcFile.Close()
+
+				_, err = io.Copy(destFile, srcFile)
+				Expect(err).NotTo(HaveOccurred(), "copy file")
+
+			}
+		})
+	})
+
+	Context("when deploying app", func() {
+		var (
+			appDeployErr error
+		)
+		BeforeAll(func() {
+			_, appDeployErr = cli.AppDeploy(&client.AppDeployInput{
+				AppName:  appName,
+				EnvName:  "test",
+				ImageTag: "gallopinggurdey",
+			})
+		})
+
+		It("app deploy should succeed", func() {
+			Expect(appDeployErr).NotTo(HaveOccurred())
+		})
+
+		It("should be able to make a POST request", func() {
+			app, appShowErr := cli.AppShow(&client.AppShowRequest{
+				ProjectName: projectName,
+				AppName:     appName,
+			})
+			Expect(appShowErr).NotTo(HaveOccurred())
+			Expect(len(app.Routes)).To(Equal(1))
+
+			// Make a POST request to the API to store the user name in DynamoDB.
+			route := app.Routes[0]
+			Expect(route.Environment).To(Equal("test"))
+			Expect(route.URL).To(HaveSuffix(appName))
+			resp, fetchErr := http.Post(fmt.Sprintf("http://%s/efekarakus", route.URL), "application/json", nil)
```

Comment:
Should probably just rap this in an eventually

Reply:
Done